exchanges/poloniex/poloniex.go

```diff
@@ -16,887 +16,1047 @@ import (
-	poloniexOrderCancel          = "cancelOrder"
-	poloniexOrderMove            = "moveOrder"
-	poloniexWithdraw             = "withdraw"
-	poloniexFeeInfo              = "returnFeeInfo"
-	poloniexAvailableBalances    = "returnAvailableAccountBalances"
-	poloniexTradableBalances     = "returnTradableBalances"
-	poloniexTransferBalance      = "transferBalance"
-	poloniexMarginAccountSummary = "returnMarginAccountSummary"
-	poloniexMarginBuy            = "marginBuy"
-	poloniexMarginSell           = "marginSell"
-	poloniexMarginPosition       = "getMarginPosition"
-	poloniexMarginPositionClose  = "closeMarginPosition"
-	poloniexCreateLoanOffer      = "createLoanOffer"
-	poloniexCancelLoanOffer      = "cancelLoanOffer"
-	poloniexOpenLoanOffers       = "returnOpenLoanOffers"
-	poloniexActiveLoans          = "returnActiveLoans"
-	poloniexLendingHistory       = "returnLendingHistory"
-	poloniexAutoRenew            = "toggleAutoRenew"
-	poloniexCancelByIDs          = "/orders/cancelByIds"
-	poloniexTimestamp            = "/timestamp"
-	poloniexWalletActivity       = "/wallets/activity"
-	poloniexMaxOrderbookDepth    = 100
+	apiURL           = "https://api.poloniex.com"
+	tradeSpotPath    = "/trade/"
+	tradeFuturesPath = "/futures" + tradeSpotPath
+	marketsPath      = "/markets/"
 )
 
-// Exchange implements exchange.IBotExchange and contains additional specific api methods for interacting with Poloniex
+var (
+	errAddressRequired         = errors.New("address is required")
+	errInvalidWithdrawalChain  = errors.New("invalid withdrawal chain")
+	errInvalidTimeout          = errors.New("invalid timeout")
+	errChainsNotFound          = errors.New("chains not found")
+	errAccountIDRequired       = errors.New("missing account ID")
+	errAccountTypeRequired     = errors.New("account type required")
+	errMarginAdjustTypeMissing = errors.New("margin adjust type invalid")
+	errPositionModeInvalid     = errors.New("invalid position mode")
+	errTrailingOffsetInvalid   = errors.New("invalid trailing offset required for trailing stop orders")
+	errOffsetLimitInvalid      = errors.New("invalid offset required for trailing stop limit orders")
+)
+
+// Exchange is the overarching type across the poloniex package
 type Exchange struct {
 	exchange.Base
-	details CurrencyDetails
 }
 
-// GetTicker returns current ticker information
-func (e *Exchange) GetTicker(ctx context.Context) (map[string]Ticker, error) {
-	type response struct {
-		Data map[string]Ticker
+// GetSymbol returns symbol and trade limit info
+func (e *Exchange) GetSymbol(ctx context.Context, symbol currency.Pair) ([]*SymbolDetails, error) {
+	if symbol.IsEmpty() {
+		return nil, currency.ErrSymbolStringEmpty
 	}
+	var resp []*SymbolDetails
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, unauthEPL, "/markets/"+symbol.String(), &resp)
+}
 
-	resp := response{}
-	path := "/public?command=returnTicker"
+// GetAllSymbols returns all symbols and their trade limits
```

Comment:
🚧 Change request:
This was a change request to rename to `GetSymbols`, and used `GetAllSymbols`.
I don't mind that you chose not to take my suggestion, **but for clarity can you reply to threads to explain your motivations.**
Right now I don't know if you just missed that I removed `All` as well as `Information`, or if you decided against it.

I still hold that we don't use `GetAll*` in other APIs, so we shouldn't here for consistency. And also because `All` is implicit.

```suggestion
// GetSymbols returns all symbols and their trade limits
func (e *Exchange) GetSymbols(ctx context.Context) ([]*SymbolDetails, error) {
```

Reply:
I accepted your suggestion but with the keyword "All" added because we may be needed to explain that this call will affect all records if zero value is provided; for all endpoint that has the keyword "All" in them.

Except for this reason, removing 'All' from these endpoints name is very easy and insignificant for me.